src/main/java/com/google/sps/servlets/PotentialMatchesServlet.java

```diff
@@ -60,28 +68,74 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   /**
-  * Loads the next potential match for a user
+  * Loads the match information for a specified user if match information is not
+  * already in datastore.
   *
-  * <p>Currently is just loading all the possible matches and obtaining one of them.
+  * <p>Creates new match information entity for user if it had not already been stored.
   *
-  * <p>TODO(#21): Obtain potential match results from datastore instead of running the 
-  * potential match finding methods repeatedly.
+  * @param userID The ID of the user who's match information is being checked/loaded into datastore
+  */
+  private void loadUserMatchInformation(String userID) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity matchInfoEntity = datastore.prepare(new Query(MATCH_INFO_ENTITY).setFilter(
+      new FilterPredicate(USER_ID_PROPERTY, FilterOperator.EQUAL, userID))).asSingleEntity();
+    
+    if (matchInfoEntity == null) {
+      addMatchInfoToDatastore(userID);
+    }
+  }
+  
+  /**
+  * Given a specific user, get their next potential match for their feed page
+  *
+  * @param userID The user whose potential match is being retrieved
+  */
+  private String getNextPotentialMatchID(String userID) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
nit: could this be declared above all the functions with the other variables?

Reply:
Yup, I can make this a final class variable.